core/vm/evm.go

```diff
@@ -47,14 +47,40 @@ type (
 
 // run runs the given contract and takes care of running precompiles with a fallback to the byte code interpreter.
 func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, error) {
-	// Intercept the StateManager calls
-	if contract.Address() == StateManagerAddress {
-		log.Debug("Calling State Manager contract.", "StateManagerAddress", StateManagerAddress.Hex())
-		ret, err := callStateManager(input, evm, contract)
-		if err != nil {
-			log.Error("State manager error!", "error", err)
+	if UsingOVM {
+		// OVM_ENABLED
+
+		// Some simple logging here. First, check to see if we know about the address we're
+		// interacting with and try to log the input data.
+		var isUnknown = true
+		for name, account := range OvmStateDump.Accounts {
+			if contract.Address() == account.Address {
+				isUnknown = false
+				abi := &(account.ABI)
+				method, err := abi.MethodById(input)
+				if err != nil {
+					log.Debug("Calling Known Contract", "Name", name, "ERROR", err)
+				} else {
+					log.Debug("Calling Known Contract", "Name", name, "Method", method.RawName)
+				}
+			}
+		}
+
+		// We don't know the contract, so print some generic information.
+		if isUnknown {
+			log.Debug("Calling Unknown Contract", "Address", contract.Address().Hex())
+		}
+
+		// Temporary: Safety checker always returns true.
+		// ovmTODO: Remove this.
+		if contract.Address() == OvmStateDump.Accounts["OVM_SafetyChecker"].Address {
+			return AbiBytesTrue, nil
+		}
+
+		// If we're calling the state manager, we want to use our native implementation instead.
+		if contract.Address() == OvmStateManager.Address {
+			return callStateManager(input, evm, contract)
```

Comment:
Just to be clear, no user willq ever _directly_ call the OvmStateManager, it will always be an internal call made by the `ExecutionManager.run`'s `ovmCALL` which will execute the ovmSolc re-written `ovmSSTORE` etc., right?

Reply:
Yep no one can directly call the state manager. This is because it's a `dead` address & therefore not able to be called directly